src/reminders/Reminder.ts

```diff
@@ -0,0 +1,77 @@
+import type { Moment } from 'moment';
+import { getSettings } from '../Config/Settings';
+
+export class ReminderSettings {
+    notificationTitle: string = 'Task Reminders';
+    dateFormat: string = 'YYYY-MM-DD';
+    dateTimeFormat: string = 'YYYY-MM-DD h:mm a';
+    dailyReminderTime: string = '9:00 am';
+    refreshInterval: number = 5 * 1000; // in Miliseconds
+
+    constructor() {}
+}
+
+export class ReminderList {
+    public reminders: Reminder[] = [];
+
+    constructor(times: Moment[] | null) {
+        times?.forEach((time) => {
+            this.reminders.push(parseMoment(time));
+        });
+    }
+
+    public toString(): string {
+        return this.reminders.map((reminder) => `${reminder.toString()}`).join(', ');
+    }
+
+    // TODO only used in ReminderDateField need way to deal with modal multiple reminders
```

Comment:
What needs to be done, to satisfy this TODO?

Reply:
Need to figure out the best way to handle multiple todos in the edit modal:
<img width="505" alt="image" src="https://user-images.githubusercontent.com/16601367/235575683-acde990c-ebcd-4bcd-9d23-0bcf116e9b1b.png">
<img width="475" alt="image" src="https://user-images.githubusercontent.com/16601367/235575811-c68dfb41-3d61-4c4d-a653-2b0a364ba700.png">

As for ReminderDateField, looking at it again, maybe you can help me understand what those are for `DueDateField, StartDateField...`? Is it only needed for sort and group by queries?

Reply:
### Request: Only allow at most 1 reminder on a task

Benefits:

- Simplifies the task parsing/writing code
- Simplifies the Edit task modal
- Simplifies the filtering code
- Reduces the number of Jest tests required

I am conscious that some aspects of naming may change, though.

- The CSS name uses `reminders` and it would logically be `reminder`

The more I think about this, the more I am inclined to only ever support a single reminder date/time value.

Reply:
While I'm open to discussing dropping multiple explicit reminders, i.e `⏲️ 2023-05-01 9:15 am, 2023-05-02 6:30 pm`. My reasoning behind having multiple reminders is for the future ability to set daily reminders on due/start date. Maybe in the form of `⏲️ 2023-05-01 9:15 am, 📅, 🛫`.  I think this would be very beneficial, at least in my workflow.

- Simplifies the task parsing/writing code
   - Not sure how much it would simply this. Mostly just removing the one foreach line in task serializer, notifications, and recurrence
- Simplifies the Edit task modal
   - I think the edit modal could be limited to a single reminder, and multiple reminders could be treated as meant for advanced users and only mentioned in the advanced section of the documentation. 
- Simplifies the filtering code
   - See the filtering comment https://github.com/obsidian-tasks-group/obsidian-tasks/pull/1925#discussion_r1182246929
- Reduces the number of Jest tests required
   - fair point, would probably double the amount of tests. One for a single reminder and another for multiple 


Looking at Files changed, the bulk of the line changes Notifications.ts, Reminder modal, and Reminder.ts none of which would change that greatly by switching to a single reminder. Moving to a single reminder doesn't seem like it would save more than ~50 lines of code at the current state, most saving would probably be in reduced number of tests.

Reply:
Ah, that tells me I haven't explained my thinking very well.

It's less about removing the code you have written, and more about not having to do the things you have not yet done.

I haven't given any feedback on the missing Jest tests yet, as I thought it was better to give feedback on the functionality.

Thinking about the reading and writing of 1 reminder value, what tests would you write?

And then, if multiple values are supported, what tests would you write?

Reply:
OK, I understand you're keen to keep multiple reminders in.

So let's stick with them, and instead move to working on the unit tests and the user documentation for it.

So that means working through these two pages and doing all the steps in them please:

https://publish.obsidian.md/tasks-contributing/Code/How+do+I+add+a+new+field+to+the+Task+class

https://publish.obsidian.md/tasks-contributing/Code/How+do+I+add+a+new+task+filter

In the documentation, please be sure to record any limitations clearly.

For example:

> warning: if you have multiple reminders on your task and you edit it via the Add and Edit Task modal, all but your first reminder will be silently deleted when the task is updated.

Reply:
My first thought is you have two key tests, serializing and deserializing a task. You have 3-4 tests for each

1)  task with no reminders
2)  task with a single reminder
3)  task with multiple reminder
4)  task with multiple reminder, different styles

The multiple reminder test (3) would just use a loop to check each reminder, similar to the test for a single reminder (2). As for 4) there are only three possible input formats "YYYY-MM-DD", "YYYY-MM-DD h:mm a", and "YYYY-MM-DD HH:mm". 

I don't think testing the Task aspect of this to be that cumbersome (will look at adding more tests this weekend). I'm envisioning creating tests for notifications to be the real tricky part.

Reply:
Please also bear in mind that if Tasks allows multiple reminders, then other plugins that read multiple reminders are also implicitly going to need to do so as well.

So it's having a broader impact than just on this project.

Reply:
> (will look at adding more tests this weekend)

Super - many thanks.